### PR TITLE
chore(hapi): database persistence of blockchain events

### LINKED ISSUE #30: Database persistence of blockchain events
verify:

- [x] #13 
- [x] #31 
- [x] #32 

They are working correctly

### PR DESCRIPTION
**Pull request by warnerHurtado**
### What does this PR do?

- Resolve #30 

### Steps to test

1. Run the project
2. Go to the folder hapi/src/services/hyperion/index and see the changes
3. Go to the folder hapi/src/services/hyperion/updaters and see the changes on eosiotoken and genesiseden

#### CheckList

- [x] Follow proper Markdown format
- [x] The content is adequate
- [x] The content is available in both english and spanish
- [x] I Ran a spell check

### CODE REVIEW

File: hapi/src/services/hyperion/updaters/eosiotoken-transfer.updater.js [hunk 1]
```diff
@@ -4,7 +4,23 @@ module.exports = {
...
-      // await edenTransactionGql.save({})
+      const transactionData = {
+        txid: action.transaction_id,
+        amount: action.data.amount,
+        category: action.data.memo, //TODO: split this from memo
+        date: action.timestamp,
+        description: action.data.memo, //TODO: split this from memo
+        id_election: action.idElection,
+        recipient: action.data.to,
+        type: 'income'   //TODO: ASK FOR THIS ONE
```
Comment (xavier506) [inline hapi/src/services/hyperion/updaters/eosiotoken-transfer.updater.js:15]: lets remove the comments

File: hapi/src/services/hyperion/updaters/genesiseden-withdraw.updater.js [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+      console.log(action); 
+      const transactionData = {
+        txid: action.transaction_id,
+        amount: action.data.amount,
+        category: action.data.memo, //TODO: split this from memo
+        date: action.timestamp,
+        description: action.data.memo, //TODO: split this from memo
+        id_election: action.idElection,
+        recipient: action.data.to,
+        type: 'expense' //TODO: ASK FOR THIS ONE
```
Comment (xavier506) [inline hapi/src/services/hyperion/updaters/genesiseden-withdraw.updater.js:16]: remove comments

File: hapi/src/services/hyperion/index.js [hunk 3]
```diff
@@ -7,6 +7,8 @@ const hyperionStateService = require('../../gql/hyperion-state.gql')
 
 const updaters = require('./updaters')
```
Comment (leisterfrancisco) [inline hapi/src/services/hyperion/index.js:9]: remove blank line

File: hapi/src/services/hyperion/index.js [hunk 4]
```diff
@@ -7,6 +7,8 @@ const hyperionStateService = require('../../gql/hyperion-state.gql')
```
Comment (leisterfrancisco) [inline hapi/src/services/hyperion/index.js:7]: remove blank line

File: hapi/src/services/hyperion/index.js [hunk 5]
```diff
@@ -92,6 +94,7 @@ const getActions = async params => {
 }
 
 const runUpdaters = async actions => {
+
```
Comment (leisterfrancisco) [inline hapi/src/services/hyperion/index.js:97]: remove blank line

File: hapi/src/services/hyperion/index.js [hunk 6]
```diff
@@ -102,7 +105,11 @@ const runUpdaters = async actions => {
       continue
     }
```
Comment (leisterfrancisco) [inline hapi/src/services/hyperion/index.js:107]: clean code: remove blank line

File: hapi/src/services/hyperion/index.js [hunk 7]
```diff
@@ -125,8 +132,9 @@ const sync = async () => {
 
   try {
     while (hasMore) {
-      ;({ hasMore, actions } = await getActions({ after, before, skip }))
+      ; ({ hasMore, actions } = await getActions({ after, before, skip }))
```
Comment (leisterfrancisco) [inline hapi/src/services/hyperion/index.js:135]: why?

File: hapi/src/services/hyperion/updaters/genesiseden-withdraw.updater.js [hunk 8]
```diff
@@ -1,7 +1,27 @@
+const { edenTransactionGql } = require('../../../gql')
+
 module.exports = {
   type: `genesis.eden:withdraw`,
   apply: async action => {
     try {
+      console.log(action);
```
Comment (leisterfrancisco) [inline hapi/src/services/hyperion/updaters/genesiseden-withdraw.updater.js:7]: is this console log necessary?

File: hapi/src/services/hyperion/updaters/genesiseden-withdraw.updater.js [hunk 9]
```diff
@@ -1,7 +1,27 @@
...
+        recipient: action.data.to,
+        type: 'expense' //TODO: ASK FOR THIS ONE
+      }
+
+      const registeredTransaction = await edenTransactionGql.get({
+        txid: { _eq: transactionData.txid }
+      });      
+
+      if (!registeredTransaction) await edenTransactionGql.save(transactionData)
+
```
Comment (leisterfrancisco) [inline hapi/src/services/hyperion/updaters/genesiseden-withdraw.updater.js:24]: remove blank line

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

**Comment by warnerHurtado** [comment]
Requested changes fixed.

**Comment by xavier506** [comment]
> This pull request **fixes 1 alert** when merging [afc0f81](https://github.com/eoscostarica/eosio-spend-explorer/commit/afc0f81ca611b1c6250a9a369b561c1f9b7294f5) into [9cce1c3](https://github.com/eoscostarica/eosio-spend-explorer/commit/9cce1c31e6e95f25b07afc671659a67421370906) - [view on LGTM.com](https://lgtm.com/projects/g/eoscostarica/eosio-spend-explorer/rev/pr-15fe271ef679aff9fceedf10a99167aaeed09862)
> 
> **fixed alerts:**
> 
> * 1 for Unused variable, import, function or class

@warnerHurtado  please check

image

**Review by xavier506 — APPROVED** [review]
LGTM

**Pull request merged**